src/files/XmlFile.ts

```diff
@@ -369,6 +370,38 @@ export class XmlFile {
         return null;
     }
 
+    public getCodeActions(range: Range) {
+        const result = [] as CodeAction[];
+
+        for (const diagnostic of this.diagnostics) {
+            //skip diagnostics that don't occur on this line
+            if (diagnostic.range?.start.line !== range.start.line) {
+                continue;
+            }
+            if (diagnostic.code === DiagnosticMessages.xmlComponentMissingExtendsAttribute().code) {
+                //add the attribute at the end of the first attribute, or after the `<component` if no attributes
+                const pos = (this.parser.ast.component.attributes[0] ?? this.parser.ast.component.tag).range.end;
+                //add this extends attribute right after the closing `>` token
+                range.end.character -= 1;
```

Comment:
Don't mutate input data.

Reply:
good catch. I wasn't actually even using this anymore. Removed.